### PR TITLE
Implement secondaries stat chance overflow

### PR DESCRIPTION
**Pull request by DaWoblefet**
For stat increases, and plausibly stat drops, secondaries will overflow when both Serene Grace and Pledge Rainbow are simultaneously active. This is because the variable that stores the value is an unsigned 8-bit integer. The only move this affects is Charge Beam. See the [battle mechanics research thread](https://www.smogon.com/forums/posts/9076566/) for more details on its discovery.

In battle-actions, in addition to the ``secondaries`` function, there's another function called ``selfDrops`` that also checks for secondary chances as far as I can tell, but it seems to be unused throughout the codebase. I have no idea what it does but don't think it needs to be touched here.

### CODE REVIEW

File: test/sim/abilities/serenegrace.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+const assert = require('./../../assert');
+const common = require('./../../common');
+
+let battle;
+
+describe(`Serene Grace`, function () {
+	afterEach(function () {
+		battle.destroy();
+	});
+
```
Comment (Slayer95) [inline test/sim/abilities/serenegrace.js:12]: This needs an ``after()`` hook to restore the secondary chances when the test is done. Also it would be the best practice to accordingly mod them in a ``before()`` hook.

**Comment by Slayer95** [comment]
``selfDrops`` seems to be for Close Combat et. al., and, relevantly, Diamond Storm?

Also, ``56ba7829661`` removed a debug instruction from battle.random()
``this.debug('randBW(' + (m ? (n ? m + ', ' + n : m) : '') + ') = ' + result)``

In hindsight, adding that back would allow for a test easily checking for the actual overflow in Charge Beam, and would in general improve many prng related tests, wdyt?

File: test/sim/abilities/serenegrace.js (hunk 1 above)
Comment (DaWoblefet) [reply test/sim/abilities/serenegrace.js:12]: Very good catch! I resolved this by editing the secondary mid-event rather than modifying the actual move in Dex.

**Comment by DaWoblefet** [comment]
> `selfDrops` seems to be for Close Combat et. al., and, relevantly, Diamond Storm?

It's not for Diamond Storm, because Diamond Storm is a secondary (and so ``isSecondary`` is true and RNG isn't called). Close Combat would be example. I can't think of a percentage-based effect offhand like that has boosts but isn't a secondary, though.

> Also, [56ba782](https://github.com/smogon/pokemon-showdown/commit/56ba78296614b185d3485229c5885ad6442e4a42) removed a debug instruction from `battle.random()` `this.debug('randBW(' + (m ? (n ? m + ', ' + n : m) : '') + ') = ' + result)`
> 
> In hindsight, adding that back would allow for a test easily checking for the actual overflow in Charge Beam, and would in general improve many prng related tests, wdyt?

I don't understand what it was like previously so I can't say. If I can help it, I prefer not involving RNG in tests at all, but if you have to, I don't mind just fooling with RNG seeds until we get one that works. Fwiw you can't actually check for debug

**Comment by Slayer95** [comment]
That's not what I was suggesting. Anyway, I realized my actual idea was impracticable, so please disregard that comment.

This now looks good ~

File: test/sim/abilities/serenegrace.js (hunk 1 above)
Comment (Slayer95) [reply test/sim/abilities/serenegrace.js:12]: Right, that's a better approach.
